src/main/java/com/devcycle/sdk/server/openfeature/DevCycleProvider.java

```
@@ -0,0 +1,144 @@
+                throw new TypeMismatchError("DevCycle JSON objects may only contain strings, numbers, booleans and nulls");
+            }
+        }
+
+        if (devcycleClient.isInitialized()) {
+            try {
+                DevCycleUser user = DevCycleUser.fromEvaluationContext(ctx);
+
+                Variable<Object> variable = devcycleClient.variable(user, key, defaultValue.asStructure().asObjectMap());
+
+                if (variable == null || variable.getIsDefaulted()) {
+                    return ProviderEvaluation.<Value>builder()
+                            .value(defaultValue)
+                            .reason(Reason.DEFAULT.toString())
+                            .build();
+                } else {
+                    if (variable.getValue() instanceof Map) {
+                        // JSON objects are managed as Map implementations and must be converted to an OpenFeature structure
+                        Value objectValue = new Value(Structure.mapToStructure((Map) variable.getValue()));
+                        return ProviderEvaluation.<Value>builder()
+                                .value(objectValue)
+                                .reason(Reason.TARGETING_MATCH.toString())
+                                .build();
+                    } else {
+                        throw new TypeMismatchError("DevCycle variable for key " + key + " is not a JSON object");
+                    }
+                }
+            } catch (IllegalArgumentException e) {
+                return ProviderEvaluation.<Value>builder()
+                        .value(defaultValue)
+                        .reason(Reason.ERROR.toString())
+                        .errorCode(ErrorCode.GENERAL)
+                        .errorMessage(e.getMessage())
+                        .build();
+            }
+        } else {
+            throw new ProviderNotReadyError("DevCycle client not initialized");
```

Do we not want to return a default when the SDK isn't initialized yet?

---

So remember that the provider is used by the OpenFeature API, so it will swallow this exception and set the appropriate Error message and reason code.

Check out https://github.com/open-feature/java-sdk/blob/7f66afa8e67d19a5169a2d07e46d2117d810dbd2/src/main/java/dev/openfeature/sdk/OpenFeatureClient.java#L139C11-L139C11